### ISSUE TITLE
Ioda-converters YAML for NEXRAD data

### ISSUE THREAD

**Issue by PraveenKumar-NOAA** [NOAA-EMC/RDASApp#58]
# Description 
Write an example YAML for the NEXRAD 3[D]-RTMA/RRFS data that converts a BUFR file to an IODA file using the bufr2ioda.x for the following subsets:

| MNEMONIC | NUMBER | DESCRIPTION                                              |
|----------|--------|----------------------------------------------------------|
|          |        |                                                          |
| NC006001 | A63243 | M TYPE 006-001 NEXRAD LVL 3 SUPEROB RADIAL WIND          |
| NC006002 | A62244 | M TYPE 006-002 NEXRAD LVL 2.5 SUPEROB RADIAL WIND        |
| NC006010 | A58010 | M TYPE 006-010 NEXRAD LVL 2 RADIAL WIND (00z)            |
| NC006011 | A58011 | M TYPE 006-011 NEXRAD LVL 2 RADIAL WIND (01z)            |
| NC006012 | A58012 | M TYPE 006-012 NEXRAD LVL 2 RADIAL WIND (02z)            |
| NC006013 | A58013 | M TYPE 006-013 NEXRAD LVL 2 RADIAL WIND (03z)            |
| NC006014 | A58014 | M TYPE 006-014 NEXRAD LVL 2 RADIAL WIND (04z)            |
| NC006015 | A58015 | M TYPE 006-015 NEXRAD LVL 2 RADIAL WIND (05z)            |
| NC006016 | A58016 | M TYPE 006-016 NEXRAD LVL 2 RADIAL WIND (06z)            |
| NC006017 | A58017 | M TYPE 006-017 NEXRAD LVL 2 RADIAL WIND (07z)            |
| NC006018 | A58018 | M TYPE 006-018 NEXRAD LVL 2 RADIAL WIND (08z)            |
| NC006019 | A58019 | M TYPE 006-019 NEXRAD LVL 2 RADIAL WIND (09z)            |
| NC006020 | A58020 | M TYPE 006-020 NEXRAD LVL 2 RADIAL WIND (10z)            |
| NC006021 | A58021 | M TYPE 006-021 NEXRAD LVL 2 RADIAL WIND (11z)            |
| NC006022 | A58022 | M TYPE 006-022 NEXRAD LVL 2 RADIAL WIND (12z)            |
| NC006023 | A58023 | M TYPE 006-023 NEXRAD LVL 2 RADIAL WIND (13z)            |
| NC006024 | A58024 | M TYPE 006-024 NEXRAD LVL 2 RADIAL WIND (14z)            |
| NC006025 | A58025 | M TYPE 006-025 NEXRAD LVL 2 RADIAL WIND (15z)            |
| NC006026 | A58026 | M TYPE 006-026 NEXRAD LVL 2 RADIAL WIND (16z)            |
| NC006027 | A58027 | M TYPE 006-027 NEXRAD LVL 2 RADIAL WIND (17z)            |
| NC006028 | A58028 | M TYPE 006-028 NEXRAD LVL 2 RADIAL WIND (18z)            |
| NC006029 | A58029 | M TYPE 006-029 NEXRAD LVL 2 RADIAL WIND (19z)            |
| NC006030 | A58030 | M TYPE 006-030 NEXRAD LVL 2 RADIAL WIND (20z)            |
| NC006031 | A58031 | M TYPE 006-031 NEXRAD LVL 2 RADIAL WIND (21z)            |
| NC006032 | A58032 | M TYPE 006-032 NEXRAD LVL 2 RADIAL WIND (22z)            |
| NC006033 | A58033 | M TYPE 006-033 NEXRAD LVL 2 RADIAL WIND (23z)            |
| NC006040 | A58040 | M TYPE 006-040 NEXRAD LVL 2 REFLECTIVITY (00z)           |
| NC006041 | A58041 | M TYPE 006-041 NEXRAD LVL 2 REFLECTIVITY (01z)           |
| NC006042 | A58042 | M TYPE 006-042 NEXRAD LVL 2 REFLECTIVITY (02z)           |
| NC006043 | A58043 | M TYPE 006-043 NEXRAD LVL 2 REFLECTIVITY (03z)           |
| NC006044 | A58044 | M TYPE 006-044 NEXRAD LVL 2 REFLECTIVITY (04z)           |
| NC006045 | A58045 | M TYPE 006-045 NEXRAD LVL 2 REFLECTIVITY (05z)           |
| NC006046 | A58046 | M TYPE 006-046 NEXRAD LVL 2 REFLECTIVITY (06z)           |
| NC006047 | A58047 | M TYPE 006-047 NEXRAD LVL 2 REFLECTIVITY (07z)           |
| NC006048 | A58048 | M TYPE 006-048 NEXRAD LVL 2 REFLECTIVITY (08z)           |
| NC006049 | A58049 | M TYPE 006-049 NEXRAD LVL 2 REFLECTIVITY (09z)           |
| NC006050 | A58050 | M TYPE 006-050 NEXRAD LVL 2 REFLECTIVITY (10z)           |
| NC006051 | A58051 | M TYPE 006-051 NEXRAD LVL 2 REFLECTIVITY (11z)           |
| NC006052 | A58052 | M TYPE 006-052 NEXRAD LVL 2 REFLECTIVITY (12z)           |
| NC006053 | A58053 | M TYPE 006-053 NEXRAD LVL 2 REFLECTIVITY (13z)           |
| NC006054 | A58054 | M TYPE 006-054 NEXRAD LVL 2 REFLECTIVITY (14z)           |
| NC006055 | A58055 | M TYPE 006-055 NEXRAD LVL 2 REFLECTIVITY (15z)           |
| NC006056 | A58056 | M TYPE 006-056 NEXRAD LVL 2 REFLECTIVITY (16z)           |
| NC006057 | A58057 | M TYPE 006-057 NEXRAD LVL 2 REFLECTIVITY (17z)           |
| NC006058 | A58058 | M TYPE 006-058 NEXRAD LVL 2 REFLECTIVITY (18z)           |
| NC006059 | A58059 | M TYPE 006-059 NEXRAD LVL 2 REFLECTIVITY (19z)           |
| NC006060 | A58060 | M TYPE 006-060 NEXRAD LVL 2 REFLECTIVITY (20z)           |
| NC006061 | A58061 | M TYPE 006-061 NEXRAD LVL 2 REFLECTIVITY (21z)           |
| NC006062 | A58062 | M TYPE 006-062 NEXRAD LVL 2 REFLECTIVITY (22z)           |
| NC006063 | A58063 | M TYPE 006-063 NEXRAD LVL 2 REFLECTIVITY (23z)           |
| NC006070 | A58070 | M TYPE 006-070 P-3 ACFT TAIL DOPPLER RADAR RADIAL WIND   |
| NC006080 | A58080 | M TYPE 006-080 CANADIAN RADAR REFLECTIVITY (00z)         |
| NC006081 | A58081 | M TYPE 006-081 CANADIAN RADAR REFLECTIVITY (01z)         |
| NC006082 | A58082 | M TYPE 006-082 CANADIAN RADAR REFLECTIVITY (02z)         |
| NC006083 | A58083 | M TYPE 006-083 CANADIAN RADAR REFLECTIVITY (03z)         |
| NC006084 | A58084 | M TYPE 006-084 CANADIAN RADAR REFLECTIVITY (04z)         |
| NC006085 | A58085 | M TYPE 006-085 CANADIAN RADAR REFLECTIVITY (05z)         |
| NC006086 | A58086 | M TYPE 006-086 CANADIAN RADAR REFLECTIVITY (06z)         |
| NC006087 | A58087 | M TYPE 006-087 CANADIAN RADAR REFLECTIVITY (07z)         |
| NC006088 | A58088 | M TYPE 006-088 CANADIAN RADAR REFLECTIVITY (08z)         |
| NC006089 | A58089 | M TYPE 006-089 CANADIAN RADAR REFLECTIVITY (09z)         |
| NC006090 | A58090 | M TYPE 006-090 CANADIAN RADAR REFLECTIVITY (10z)         |
| NC006091 | A58091 | M TYPE 006-091 CANADIAN RADAR REFLECTIVITY (11z)         |
| NC006092 | A58092 | M TYPE 006-092 CANADIAN RADAR REFLECTIVITY (12z)         |
| NC006093 | A58093 | M TYPE 006-093 CANADIAN RADAR REFLECTIVITY (13z)         |
| NC006094 | A58094 | M TYPE 006-094 CANADIAN RADAR REFLECTIVITY (14z)         |
| NC006095 | A58095 | M TYPE 006-095 CANADIAN RADAR REFLECTIVITY (15z)         |
| NC006096 | A58096 | M TYPE 006-096 CANADIAN RADAR REFLECTIVITY (16z)         |
| NC006097 | A58097 | M TYPE 006-097 CANADIAN RADAR REFLECTIVITY (17z)         |
| NC006098 | A58098 | M TYPE 006-098 CANADIAN RADAR REFLECTIVITY (18z)         |
| NC006099 | A58099 | M TYPE 006-099 CANADIAN RADAR REFLECTIVITY (19z)         |
| NC006100 | A58100 | M TYPE 006-100 CANADIAN RADAR REFLECTIVITY (20z)         |
| NC006101 | A58101 | M TYPE 006-101 CANADIAN RADAR REFLECTIVITY (21z)         |
| NC006102 | A58102 | M TYPE 006-102 CANADIAN RADAR REFLECTIVITY (22z)         |
| NC006103 | A58103 | M TYPE 006-103 CANADIAN RADAR REFLECTIVITY (23z)         |
| NC006110 | A58110 | M TYPE 006-110 CANADIAN RADAR RADIAL WIND (00z)          |
| NC006111 | A58111 | M TYPE 006-111 CANADIAN RADAR RADIAL WIND (01z)          |
| NC006112 | A58112 | M TYPE 006-112 CANADIAN RADAR RADIAL WIND (02z)          |
| NC006113 | A58113 | M TYPE 006-113 CANADIAN RADAR RADIAL WIND (03z)          |
| NC006114 | A58114 | M TYPE 006-114 CANADIAN RADAR RADIAL WIND (04z)          |
| NC006115 | A58115 | M TYPE 006-115 CANADIAN RADAR RADIAL WIND (05z)          |
| NC006116 | A58116 | M TYPE 006-116 CANADIAN RADAR RADIAL WIND (06z)          |
| NC006117 | A58117 | M TYPE 006-117 CANADIAN RADAR RADIAL WIND (07z)          |
| NC006118 | A58118 | M TYPE 006-118 CANADIAN RADAR RADIAL WIND (08z)          |
| NC006119 | A58119 | M TYPE 006-119 CANADIAN RADAR RADIAL WIND (09z)          |
| NC006120 | A58120 | M TYPE 006-120 CANADIAN RADAR RADIAL WIND (10z)          |
| NC006121 | A58121 | M TYPE 006-121 CANADIAN RADAR RADIAL WIND (11z)          |
| NC006122 | A58122 | M TYPE 006-122 CANADIAN RADAR RADIAL WIND (12z)          |
| NC006123 | A58123 | M TYPE 006-123 CANADIAN RADAR RADIAL WIND (13z)          |
| NC006124 | A58124 | M TYPE 006-124 CANADIAN RADAR RADIAL WIND (14z)          |
| NC006125 | A58125 | M TYPE 006-125 CANADIAN RADAR RADIAL WIND (15z)          |
| NC006126 | A58126 | M TYPE 006-126 CANADIAN RADAR RADIAL WIND (16z)          |
| NC006127 | A58127 | M TYPE 006-127 CANADIAN RADAR RADIAL WIND (17z)          |
| NC006128 | A58128 | M TYPE 006-128 CANADIAN RADAR RADIAL WIND (18z)          |
| NC006129 | A58129 | M TYPE 006-129 CANADIAN RADAR RADIAL WIND (19z)          |
| NC006130 | A58130 | M TYPE 006-130 CANADIAN RADAR RADIAL WIND (20z)          |
| NC006131 | A58131 | M TYPE 006-131 CANADIAN RADAR RADIAL WIND (21z)          |
| NC006132 | A58132 | M TYPE 006-132 CANADIAN RADAR RADIAL WIND (22z)          |
| NC006133 | A58133 | M TYPE 006-133 CANADIAN RADAR RADIAL WIND (23z)          |
 

- [x] Extract the BUFR table, queries, and important Mnemonics information
- [ ] Use this YAML to convert into an IODA-formatted netCDF file
- [ ] Ioda obs validation using ioda-validate.x
- [ ] Perform validation for the observations using the input BUFR and output IODA file
- [ ] Add output(s) to the testoutput directory
- [ ] Add tests to the CMakeLists.txt files
- [ ] Have the tests pass

# Requirements 
All important mnemonics required by the [3D]-RTMA/RRFS users need to be known.

# Acceptance Criteria 
It will be finished when all files have been checked and tests have passed.

# Dependencies
The bufr2ioda.x

**Comment by PraveenKumar-NOAA** [comment]
Performed validation of the ioda-obs in the output ioda file generated using the bufr2ioda-x and following results were obtained:

**Comment by PraveenKumar-NOAA** [comment]
Validation of the output IODA results, obtained using the nexrad ioda-converters YAML and bufr2ioda.x, is performed for the following two observations:

**unfoldingVelocity (HNQV)
BUFR plot:**

![bufr_nexrad_unfoldingVelocity](https://github.com/NOAA-EMC/RDASApp/assets/63739712/7507ece0-d12f-4aa2-ad1b-aa6f84d50cc7)

**IODA plot:**
![nexrad-ioda_unfoldingVelocity](https://github.com/NOAA-EMC/RDASApp/assets/63739712/423964b1-33e7-435d-8a30-0413dbb6518d)



**volumeCoveragePattern (VOCP)
BUFR plot:**
![bufr_nexrad_volumeCoveragePattern](https://github.com/NOAA-EMC/RDASApp/assets/63739712/b67d40c0-102d-44f2-8fbb-162990db9f60)



**IODA plot:*
![nexrad-ioda_volumeCoveragePattern (7)](https://github.com/NOAA-EMC/RDASApp/assets/63739712/07883165-6b31-4445-8321-14a5d1e20b9d)

**Comment by PraveenKumar-NOAA** [comment]
@ShunLiu-NOAA I dumped a few rrfs nexrad tests data files for 20220526/27 and I found the following subsets in them: 
NC006010,  NC006011,  NC006012,  NC006013,  NC006014,  NC006015,  
NC006027,  NC006028,  NC006029,  NC006033.

Details about these subsets are given above. During our meeting yesterday you mentioned something about the CANADIAN data, let me know which one you/ RRFS users are interested?
